src/platform/nrfconnect/Reboot.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "Reboot.h"
+
+#include <lib/support/TypeTraits.h>
+
+#include <hal/nrf_power.h>
+#include <zephyr/sys/reboot.h>
+
+namespace chip {
+namespace DeviceLayer {
+
+constexpr uint8_t EncodeReason(SoftwareRebootReason reason)
+{
+    constexpr uint8_t kCustomReasonFlag = 0x80;
+
+    return to_underlying(reason) | kCustomReasonFlag;
+}
+
+void Reboot(SoftwareRebootReason reason)
+{
+    const uint8_t retainedReason = EncodeReason(reason);
+
+    // The parameter passed to sys_reboot() is retained in GPREGRET (general-purpose
+    // retention register) on nRF52 SOC family, but nRF53 ignores the parameter, so
```

Comment:
Can we raise the issue about it for community?

Reply:
Yes, I can. Do you mean on teams or a JIRA task for someone?

Reply:
I think JIRA works best. Thank you!